packages/builder/src/builtin-tasks/run.ts

```diff
@@ -17,23 +23,40 @@ function filterNonExistent (scripts: string[]): string[] {
   return scripts.filter(script => !fsExtra.pathExistsSync(script));
 }
 
-export async function runMultipleScripts (runtimeEnv: AlgobRuntimeEnv,
+export async function runMultipleScripts (
+  runtimeEnv: AlgobRuntimeEnv,
   scriptNames: string[],
-  runScriptFn: (
-    relativeScriptPath: string,
-    runtimeEnv: AlgobRuntimeEnv
-  ) => Promise<void>): Promise<void> {
+  onSuccessFn: (cpData: CheckpointData, relativeScriptPath: string) => void,
+  force: boolean,
+  logTag: string,
+  wrapDeployer: (orig: AlgobDeployer) => AlgobDeployer): Promise<void> {
+  const log = debug(logTag);
+  const cpData: CheckpointData = loadCheckpointsRecursive();
+  const deployer: AlgobDeployer = wrapDeployer(new AlgobDeployerImpl(runtimeEnv, cpData));
```

Comment:
why we need a function here? This looks very hacky.

Reply:
I think that instead of wrapping, we should have a builder method which will construct right `Deployer`, Example:
```
mkDeployer(runtimeEnv, checkpoints, readOnly)
```

Reply:
The factory method that you talk about would do the same thing. Instead of a function we'd have a boolean.
Currently this doesn't do anything else so it's fine to do it this way.

Reply:
Changed to factory method